flocker/node/_deploy.py

```diff
@@ -405,13 +427,22 @@ def __init__(self, hostname, volume_service, docker_client=None,
         self.network = network
         self.volume_service = volume_service
 
-    def discover_local_state(self):
+    def discover_local_state(self, cluster_state):
         """
         List all the ``Application``\ s running on this node.
 
-        :returns: A ``Deferred`` which fires with a ``NodeState``
+        :param Deployment cluster_state: The last known state of the whole
+            cluster.
+
+        :returns: A ``Deferred`` which fires with a ``NodeApplications``
             instance.
         """
+        # XXX delete manifestations code that is unrelated to
+        # applications, return NodeApplications instance.  Application's
+        # attached volumes/manifestations can be calculated based on
+        # passed-in cluster_state. This has a potential race condition
+        # unfortunately, but probably not an issue in practice.
```

Comment:
Restarts are fine as long as the control service is still running, since it will cache the info in memory (and someday expire it I guess).